### PR TITLE
[node-manager] Distroless

### PR DESCRIPTION
**Pull request by zuzzas**
## Description

Changed all images to use distroless image by default.

## Why do we need it, and what problem does it solve?

Using images with nothing inside them except statically linked program improves security.

## What is the expected result?

1. Machines are bootstrapped successfully:

```
root@klimentyev-master-0:~# kubectl -n d8-cloud-instance-manager get machine
NAME                                     STATUS    AGE
klimentyev-system-530a423c-64bc8-4gkzj   Running   9d
klimentyev-system-530a423c-64bc8-crk56   Running   11d
klimentyev-test-530a423c-554c8-z65cg     Running   11d
root@klimentyev-master-0:~# kubectl -n d8-cloud-instance-manager delete machine klimentyev-test-530a423c-554c8-z65cg
machine.machine.sapcloud.io "klimentyev-test-530a423c-554c8-z65cg" deleted
root@klimentyev-master-0:~# kubectl -n d8-cloud-instance-manager get machine
NAME                                     STATUS    AGE
klimentyev-system-530a423c-64bc8-4gkzj   Running   9d
klimentyev-system-530a423c-64bc8-crk56   Running   11d
klimentyev-test-530a423c-554c8-4r649     Running   121m
```

2. All components are working properly, nothing bad in logs.

```
root@klimentyev-master-0:~# kubectl -n d8-cloud-instance-manager get pod
NAME                                         READY   STATUS    RESTARTS   AGE
bashible-apiserver-bfb59d649-h9qsf           1/1     Running   0          127m
cluster-autoscaler-c9bb75846-p9wmd           2/2     Running   0          127m
early-oom-f4lc5                              2/2     Running   0          126m
early-oom-nmn9t                              2/2     Running   0          127m
early-oom-sbdt4                              2/2     Running   0          126m
early-oom-svj8d                              2/2     Running   0          119m
machine-controller-manager-8cd8dd49d-jftnj   2/2     Running   0          127m
```

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: feature
summary: All images are now based on distroless image
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/aws'...

**Comment by z9r5** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 5963392388 5963392388-1-con-1-25 /sys/deckhouse-oss/install:pr5520 centos@3.123.249.81

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/gcp'...

**Comment by zuzzas** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.25 6009989982 6009989982-1-con-1-25 /sys/deckhouse-oss/install:pr5520 user@34.89.136.83

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/gcp'...

**Comment by zuzzas** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.25 6011618786 6011618786-1-con-1-25 /sys/deckhouse-oss/install:pr5520 user@34.159.72.57

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/gcp'...

**Comment by zuzzas** [comment]
/e2e/abort gcp;WithoutNAT;containerd;1.25 6013727009 6013727009-1-con-1-25 /sys/deckhouse-oss/install:pr5520 user@34.89.133.140

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/gcp'...

File: modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml [hunk 1]
```diff
@@ -29,10 +29,10 @@ shell:
   install:
   - mkdir /src
   - cd /src
-  - wget https://github.com/gardener/autoscaler/archive/v{{ $version }}.{{ $value.clusterAutoscalerPatch }}.tar.gz -O - | tar -xz --strip-components=1
+  - git clone --depth 1 --branch v{{ $version }}.{{ $value.clusterAutoscalerPatch }} {{ env "SOURCE_REPO" | default "https://github.com" }}/gardener/autoscaler.git .
   - test -d /patches/{{ $version }} && for patchfile in /patches/{{ $version }}/*.patch ; do patch -p1 < ${patchfile}; done
   - cd cluster-autoscaler
-  - GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags="-s -w" -o cluster-autoscaler
+  - GOPROXY={{ env "GOPROXY" }} GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags="-s -w" -o cluster-autoscaler
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml:35]:
```suggestion
  - GOPROXY={{ .GOPROXY }} GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags="-s -w" -o cluster-autoscaler
```

File: modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml [hunk 2]
```diff
@@ -1,40 +1,26 @@
...
   install:
-  - name: "Build"
-    shell: make build-local
-    args:
-      chdir: /src
+  - apk add mercurial make bash git
+  setup:
+  - mkdir /src && cd /src
+  - git clone --depth 1 --branch v0.36.0-flant.13 {{ env "SOURCE_REPO" | default "https://github.com" }}/deckhouse/mcm.git .
+  - GOPROXY={{ env "GOPROXY" }} CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -mod=vendor -o machine-controller-manager cmd/machine-controller-manager/controller_manager.go
```
Comment (RomanenkoDenys) [inline modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml:24]:
```suggestion
  - GOPROXY={{ .GOPROXY }} CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -mod=vendor -o machine-controller-manager cmd/machine-controller-manager/controller_manager.go
```

**Review by RomanenkoDenys — COMMENTED** [review]
some nits

File: modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml [hunk 3]
```diff
@@ -1,40 +1,26 @@
...
-      - bash
-      - tzdata
-      update_cache: yes
+shell:
   install:
-  - name: "Build"
-    shell: make build-local
-    args:
-      chdir: /src
+  - apk add mercurial make bash git
```
Comment (aleksey-su) [inline modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml:20]: `mercurial make bash` extra packages

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/gcp'...

File: modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml (hunk 3 above)
Comment (zuzzas) [reply modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml]: Fixed.

File: modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml (hunk 2 above)
Comment (zuzzas) [reply modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml:24]: Fixed.

File: modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml (hunk 1 above)
Comment (zuzzas) [reply modules/040-node-manager/images/cluster-autoscaler/werf.inc.yaml]: Fixed.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/gcp'...

**Review by aleksey-su — DISMISSED** [review]

File: modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml [hunk 4]
```diff
@@ -19,27 +19,20 @@ git:
...
   setup:
-  - shell: go build -ldflags="-s -w" -o bashible-apiserver .
-    args:
-      chdir: /src
-    environment:
-      GOOS: "linux"
-      GOARCH: "amd64"
-      CGO_ENABLED: "0"
+  - cd /src
+  - CGO_ENABLED=0 GOPROXY={{ .GOPROXY }} GOOS=linux GOARCH=amd64 go build -ldflags="-s -w" -o bashible-apiserver .
```
Comment (name212) [inline modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml:30]: do we need a dollar here,

File: modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml [hunk 5]
```diff
@@ -1,40 +1,26 @@
...
+shell:
   install:
-  - name: "Build"
-    shell: make build-local
-    args:
-      chdir: /src
+  - apk add git
+  setup:
+  - mkdir /src && cd /src
+  - git clone --depth 1 --branch v0.36.0-flant.13 {{ env "SOURCE_REPO" | default "https://github.com" }}/deckhouse/mcm.git .
```
Comment (name212) [inline modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml:23]: env? Mabe we will use werf variable as here 
https://github.com/deckhouse/deckhouse/pull/5520/files#diff-af1683f87626a34a29eed0b462e74aa9c802c76b5242f8070dd96afc135df6d7R32
?

File: modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml [hunk 6]
```diff
@@ -19,27 +19,20 @@ git:
...
-      name: git
-      update_cache: yes
-  - command: rm -rf /var/cache/apk/*
+  - apk add git
   install:
-  - shell: go mod download
-    args:
-      chdir: /src
+  - cd /src
+  - go mod download
```
Comment (name212) [inline modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml:27]: I think better way export GOPROXY here

go mod download will use it

File: modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml (hunk 4 above)
Comment (zuzzas) [reply modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml]: We don't, but let's have it the same way everywhere.

File: modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml (hunk 5 above)
Comment (zuzzas) [reply modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml]: Fixed.

File: modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml (hunk 6 above)
Comment (zuzzas) [reply modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml]: Fixed.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @zuzzas. I've started the workflow for label 'e2e/run/gcp'...

File: modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml [hunk 7]
```diff
@@ -19,27 +19,20 @@ git:
...
-      GOARCH: "amd64"
-      CGO_ENABLED: "0"
+  - cd /src
+  - CGO_ENABLED=0 GOPROXY={{ $.GOPROXY }} GOOS=linux GOARCH=amd64 go build -ldflags="-s -w" -o bashible-apiserver .
+  - chown 64535:64535 bashible-apiserver
+  - chmod 0700 bashible-apiserver
 ---
 image: {{ .ModuleName }}/{{ .ImageName }}
-from: {{ .Images.BASE_ALPINE }}
+fromImage: common/distroless
```
Comment (yalosev) [inline modules/040-node-manager/images/bashible-apiserver/werf.inc.yaml:35]: Maybe 
```suggestion
fromImage: {{ .Images.BASE_DISTROLESS }}
```

File: modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml [hunk 8]
```diff
@@ -1,40 +1,26 @@
 image: {{ .ModuleName }}/{{ .ImageName }}
-from: {{ .Images.BASE_ALPINE }}
+fromImage: common/distroless
```
Comment (yalosev) [inline modules/040-node-manager/images/machine-controller-manager/werf.inc.yaml:2]:
```suggestion
fromImage: {{ .Images.BASE_DISTROLESS }}
```

**Review by yalosev — APPROVED** [review]

**Pull request merged**